### ISSUE TITLE
Is `scan_v1.cpp` correct?

### ISSUE THREAD

**Issue by kou1okada** [JPCERTCC/EmoCheck#9]
https://github.com/JPCERTCC/EmoCheck/blob/cf51da3e550d6e3d697a7def0add8b8a74bca2d7/emocheck/modules/scan_v1.cpp#L18-L25

This code makes
|ptr at L17|ptr at L26|keyword|
|----------|----------|-------|
|0         |0         |duke   |
|1         |1         |uke    |
|2         |2         |ke     |
|3         |3         |e      |
|4         |4         |mfidl  |
|5         |4         |mfidl  |
|6         |4         |mfidl  |
|...       |...       |...    |

Maybe he want as:
```C++
    {
        int i;
        for (i = ptr; i > 0; i--) {
            if (keywords[i] != ',') {
                continue;
            } else {
                break;
            }
        }
        ptr = i;
    }
```

Does this bug origin old emotet (- 2020/02/05)?

**Comment by t-tani** [comment]
Scan-v1 is based on the reverse-engineering result of old Emotet (- 2020/02/05).
We implemented the algorithm as-is written in the old Emotet binary. 
Of course, we don't have the source-code of Emotet; we couldn't determine this buggy code is the origin by old Emotet or other reasons such as compiler optimization or disassembler. We have tested this code in several environments and old samples and confirmed that is working. 

However, due to the Emotet update, Scan-v1 is not working anymore.
